Makefile

```diff
@@ -0,0 +1,130 @@
+PROJECT_PATH   = $(PROJECT_NAME)
+SHELL          = /bin/bash
+SOURCE_FOLDERS = $(PROJECT_PATH) apps docs examples licenses test
+PYTHON_FILES   = $(shell find $(SOURCE_FOLDERS) -type f -name "*.py" -o -name "*.pyi")
+COMMIT_HASH    = $(shell git log -1 --format=%h)
+PATH           := $(HOME)/go/bin:$(PATH)
+PYTHON         ?= $(shell command -v python3 || command -v python)
+PYTESTOPTS     ?=
+
+.PHONY: default
+default: install
+
+install:
+	$(PYTHON) -m pip install -vvv .
+
+install-editable:
+	$(PYTHON) -m pip install --upgrade pip
+	$(PYTHON) -m pip install -vvv --no-build-isolation --editable .
+
+install-e: install-editable  # alias
+
+uninstall:
+	$(PYTHON) -m pip uninstall -y $(PROJECT_NAME)
+
+build:
+	$(PYTHON) -m pip install --upgrade pip
+	$(PYTHON) -m pip install --upgrade setuptools wheel build
+	$(PYTHON) -m build
+
+# Tools Installation
+
+check_pip_install = $(PYTHON) -m pip show $(1) &>/dev/null || (cd && $(PYTHON) -m pip install $(1) --upgrade)
+check_pip_install_extra = $(PYTHON) -m pip show $(1) &>/dev/null || (cd && $(PYTHON) -m pip install $(2) --upgrade)
+
+pylint-install:
+	$(call check_pip_install_extra,pylint,pylint[spelling])
+	$(call check_pip_install,pyenchant)
+
+flake8-install:
+	$(call check_pip_install,flake8)
+	$(call check_pip_install,flake8-bugbear)
+	$(call check_pip_install,flake8-comprehensions)
+	$(call check_pip_install,flake8-docstrings)
+	$(call check_pip_install,flake8-pyi)
+	$(call check_pip_install,flake8-simplify)
+
+py-format-install:
+	$(call check_pip_install,isort)
+	$(call check_pip_install_extra,black,black[jupyter])
+
+ruff-install:
+	$(call check_pip_install,ruff)
+
+mypy-install:
+	$(call check_pip_install,mypy)
+
+pre-commit-install:
+	$(call check_pip_install,pre-commit)
+	$(PYTHON) -m pre_commit install --install-hooks
+
+docs-install:
+	$(call check_pip_install_extra,pydocstyle,pydocstyle[toml])
+	$(call check_pip_install,doc8)
+	$(call check_pip_install,sphinx)
+	$(call check_pip_install,sphinx-rtd-theme)
+	$(call check_pip_install,sphinx-autoapi)
+	$(call check_pip_install,sphinx-autobuild)
+	$(call check_pip_install,sphinx-copybutton)
+	$(call check_pip_install,sphinxcontrib-katex)
+	$(call check_pip_install,sphinxcontrib-bibtex)
+	$(call check_pip_install,sphinx-autodoc-typehints)
+	$(call check_pip_install,myst-nb)
+	$(call check_pip_install_extra,sphinxcontrib-spelling,sphinxcontrib-spelling pyenchant)
+
+pytest-install:
+	$(call check_pip_install,pytest)
+	$(call check_pip_install,pytest-cov)
+	$(call check_pip_install,pytest-xdist)
+
+test-install: pytest-install
+	$(PYTHON) -m pip install --requirement tests/requirements.txt
+
+# Python linters
+
+pylint: pylint-install
+	$(PYTHON) -m pylint $(PROJECT_PATH)
+
+flake8: flake8-install
+	$(PYTHON) -m flake8 --count --show-source --statistics
+
+py-format: py-format-install
+	$(PYTHON) -m isort --project $(PROJECT_PATH) --check $(PYTHON_FILES) && \
+	$(PYTHON) -m black --check $(PYTHON_FILES) tutorials
```

Comment:
Please revert to yapf. I do not see how black is better to the extent that we migrate to it.

Reply:
Also, inconsistency using `''` and `""`

Reply:
The only question is if you want to have you Makefile merged early on and then continue the infinite discussion about style preferences? @Benjamin-eecs

Reply:
sure, get makefile intergration merged then